foxpuppet/browser/window.py

```diff
@@ -22,23 +22,22 @@ def __init__(self, selenium, *args, **kwargs):
     def is_private(self):
         """Returns True if this is a Private Browsing window."""
 
-        self.selenium.set_context('chrome')
-        return self.selenium.execute_script("""
-                Components.utils.import("resource://gre/modules/PrivateBrowsingUtils.jsm");
+        with self.selenium.context('chrome'):
+            return self.selenium.execute_script("""
+                    Components.utils.import("resource://gre/modules/PrivateBrowsingUtils.jsm");
 
-                let chromeWindow = arguments[0].ownerDocument.defaultView;
-                return PrivateBrowsingUtils.isWindowPrivate(chromeWindow);
-            """, self._windows.window_element)
-        self.selenium.set_context('content')
+                    let chromeWindow = arguments[0].ownerDocument.defaultView;
+                    return PrivateBrowsingUtils.isWindowPrivate(chromeWindow);
+                """, self._windows.window_element)
 
     def open_window(self, private=False):
-        self.selenium.set_context('chrome')
-        self.selenium.find_element(*self._file_menu_button_locator).click()
-        with self._windows.wait_for_new_window():
-            if private:
-                self.selenium.find_element(
-                    *self._file_menu_private_window_locator).click()
-            else:
-                self.selenium.find_element(
-                    *self._file_menu_new_window_button_locator).click()
-        self.selenium.set_context('content')
+        with self.selenium.context('chrome'):
+            self.selenium.find_element(*self._file_menu_button_locator).click()
+            with self._windows.wait_for_new_window():
+                if private:
+                    self.selenium.find_element(
+                        *self._file_menu_private_window_locator).click()
+                else:
+                    self.selenium.find_element(
+                        *self._file_menu_new_window_button_locator).click()
+        return self.selenium.current_window_handle
```

Comment:
Why do we return the current window handle and not the one for the newly opened window? This causes extra work for any consumer of this method.